src/Database/Schema/PostgresSchemaDialect.php

```diff
@@ -255,15 +255,16 @@ public function convertColumnDescription(TableSchema $schema, array $row): void
      * schema will be used.
      *
      * @param string $tableName The table name to split
+     * @param array $config Additional configuration data
      * @return array A tuple of [schema, tablename]
      */
-    private function splitTablename(string $tableName): array
+    private function splitTablename(string $tableName, array $config = []): array
     {
-        $config = $this->_driver->config();
-        $schema = $config['schema'] ?? 'public';
         if (str_contains($tableName, '.')) {
             return explode('.', $tableName);
         }
+        $driverConfig = $this->_driver->config();
+        $schema = $config['schema'] ?? $driverConfig['schema'] ?? 'public';
```

Comment:
This is all we use it for?

Reply:
Yeah it is an override for the driver config. Some call sites of these methods pre-split the table name and provide the schema in $config.